### PR TITLE
test(amplify-e2e-tests): fixed the test accept new lambda function

### PR DESCRIPTION
**Pull request by ammarkarachi**
#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-e2e-tests/src/__tests__/apigw.test.ts [hunk 1]
```diff
@@ -30,7 +30,7 @@ describe('API Gateway e2e tests', () => {
     await amplifyPushAuth(projRoot);
     await addAuthWithGroupsAndAdminAPI(projRoot); // Groups: Admins, Users
     await amplifyPushAuth(projRoot);
-    await addRestApi(projRoot, { isFirstRestApi: false, path: '/foo' });
+    await addRestApi(projRoot, { isFirstRestApi: false, path: '/foo', existingLambda: true });
```
Comment (jhockett) [inline packages/amplify-e2e-tests/src/__tests__/apigw.test.ts:33]: this setting shouldn't matter because the path is set in the options.

Comment (jhockett) [reply packages/amplify-e2e-tests/src/__tests__/apigw.test.ts:33]: helper code:

```ts
  if (!isFirstRestApi) {
    chain.wait('Would you like to add a new path to an existing REST API');

    if (settings.path) {
      chain
        .sendYes()
        .wait('Select the REST API you want to update')
        .sendCarriageReturn() // Select the first REST API
        .wait('What would you like to do?')
        .sendCarriageReturn() // Add another path
        .wait('Provide a path')
        .sendLine(settings.path)
        .wait('Choose a lambda source')
        .sendKeyDown()
        .sendCarriageReturn() // Existing lambda
        .wait('Choose the Lambda function to invoke by this path')
        .sendCarriageReturn() // Pick first one
        .wait('Restrict API access')
        .sendNo() // Do not restrict access
        .wait('Do you want to add another path')
        .sendNo() // Do not add another path
        .sendEof();

      return chain.runAsync();
    } else {
      chain.sendNo();
    }
  }
  ...
  ```

Comment (ammarkarachi) [reply packages/amplify-e2e-tests/src/__tests__/apigw.test.ts:33]: https://app.circleci.com/pipelines/github/aws-amplify/amplify-cli/4955/workflows/cd148ede-ced8-499e-82ae-d0ac6caf1a4a/jobs/115006/tests#failed-test-0

File: packages/amplify-e2e-tests/src/__tests__/apigw.test.ts [hunk 2]
```diff
@@ -30,7 +30,7 @@ describe('API Gateway e2e tests', () => {
...
     await addRestApi(projRoot, { isFirstRestApi: false, restrictAccess: true, allowGuestUsers: true, hasUserPoolGroups: true });
```
Comment (jhockett) [inline packages/amplify-e2e-tests/src/__tests__/apigw.test.ts:34]: this is the line that needs to change.

Add the option `projectContainsFunctions: true`

**Pull request closed without merging**